src/main/java/com/skeleton/feed/service/PostService.java

```diff
@@ -72,11 +77,30 @@ public AddLikeResponse addLike(Long id) {
         return new AddLikeResponse(post);
     }
 
+
+    public PostDetailResponse getPostDetail(Long id) {
```

Comment:
```suggestion
    @Transactional
    public PostDetailResponse getPostDetail(Long id) {
```

`JPA`를 쓸 때, 서비스는 `트랜젝션` 단위로 묶어주셔야 나중에 해당 트랜젝션이 실패했을 때 `롤백`이 수행됩니다. 

내부 메서드인 incrementViewCount에 트랜젝션이 달려있긴 하지만, **외부 메서드인 getPostDetail에 @Transactional이 달려 있지 않으면** 프록시가 적용이 안되어서 내부 호출에 @Transactional 이 달려 있어도 **롤백이 수행되지 않**습니다. 그러니 여기에도 달아주는 게 좋을 것 같습니다. 

[@Transactional 사용 시 주의해야하는 8가지](https://flambeeyoga.tistory.com/entry/Transactional-%EC%82%AC%EC%9A%A9-%EC%8B%9C-%EC%A3%BC%EC%9D%98%EC%A0%90) 의 2번을 참고해주세요.

Reply:
자료 공유 감사합니다! 이 부분에 대해서 고민을 많이 했었습니다 
보통 조회 메소드는 **@Transactional(readOnly=true)**를 사용해야 직관적으로 알기 쉽고, **CUD 동작, 스냅샷 저장, dirty checking** 작업을 수행하지 않아 성능이 향상됩니다 하지만 이번 요구사항에서 조회 API 안에 조회 수 기능은 Update Query를 데이터베이스에 전달해야 합니다 
그래서 위와 같은 코드 형식을 작성했습니다만 혜리님 말씀대로 프록시가 내부 메소드인 incrementViewCount 호출을 가로채서 트랜잭션 동작을 적용할 수 없기 때문에 별도의 트랜잭션 환경에서 수행될 수 있도록 변경하겠습니다.